src/Sylius/Bundle/CoreBundle/DataFixtures/Factory/ShopUserFactory.php

```diff
@@ -0,0 +1,153 @@
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Zenstruck\Foundry\ModelFactory;
+use Zenstruck\Foundry\Proxy;
+
+/**
+ * @extends ModelFactory<ShopUserInterface>
+ *
+ * @method static ShopUserInterface|Proxy createOne(array $attributes = [])
+ * @method static ShopUserInterface[]|Proxy[] createMany(int $number, array|callable $attributes = [])
+ * @method static ShopUserInterface|Proxy find(object|array|mixed $criteria)
+ * @method static ShopUserInterface|Proxy findOrCreate(array $attributes)
+ * @method static ShopUserInterface|Proxy first(string $sortedField = 'id')
+ * @method static ShopUserInterface|Proxy last(string $sortedField = 'id')
+ * @method static ShopUserInterface|Proxy random(array $attributes = [])
+ * @method static ShopUserInterface|Proxy randomOrCreate(array $attributes = [])
+ * @method static ShopUserInterface[]|Proxy[] all()
+ * @method static ShopUserInterface[]|Proxy[] findBy(array $attributes)
+ * @method static ShopUserInterface[]|Proxy[] randomSet(int $number, array $attributes = [])
+ * @method static ShopUserInterface[]|Proxy[] randomRange(int $min, int $max, array $attributes = [])
+ * @method ShopUserInterface|Proxy create(array|callable $attributes = [])
+ */
+final class ShopUserFactory extends ModelFactory implements ShopUserFactoryInterface
+{
+    public function __construct(
+        private FactoryInterface $shopUserFactory,
+        private FactoryInterface $customerFactory,
+        private CustomerGroupFactoryInterface $customerGroupFactory,
+    ) {
+        parent::__construct();
+    }
+
+    public function withEmail(string $email): self
+    {
+        return $this->addState(['email' => $email]);
+    }
+
+    public function withFirstName(string $firstName): self
+    {
+        return $this->addState(['first_name' => $firstName]);
+    }
+
+    public function withLastName(string $lastName): self
+    {
+        return $this->addState(['last_name' => $lastName]);
+    }
+
+    public function male(): self
+    {
+        return $this->addState(['gender' => CustomerInterface::MALE_GENDER]);
+    }
+
+    public function female(): self
+    {
+        return $this->addState(['gender' => CustomerInterface::FEMALE_GENDER]);
+    }
+
+    public function withPhoneNumber(string $phoneNumber): self
+    {
+        return $this->addState(['phone_number' => $phoneNumber]);
+    }
+
+    public function withBirthday(\DateTimeInterface|string $birthday): self
+    {
+        return $this->addState(function () use ($birthday): array {
+            if (is_string($birthday)) {
+                return ['birthday' => new \DateTimeImmutable($birthday)];
+            }
+
+            return ['birthday' => $birthday];
+        });
+    }
+
+    public function withPassword(string $password): self
+    {
+        return $this->addState(['password' => $password]);
+    }
+
+    public function withCustomerGroup(Proxy|CustomerGroupInterface|string $customerGroup): self
```

Comment:
Is the `Proxy` type-hint really necessary? The proxy should implement `CustomerGroupInterface` and should pass the type check.

Reply:
Argument #1 ($customerGroup) must be of type Sylius\Component\Customer\Model\CustomerGroupInterface|string, Zenstruck\Foundry\Proxy given
So it does not implement the interface.